neps/nep-0568.md

```diff
@@ -143,6 +143,10 @@ supporting smooth transitions without altering storage structures directly.
 
 ### ShardId Semantics
 
+Currently, shard IDs are represented as numbers within the range `[0,n)`, where n is the total number of shards. These shard IDs are sorted in the same order as the account ID ranges assigned to them. While this approach is straightforward, it complicates resharding operations, particularly when splitting a shard in the middle of the range. Such a split requires reindexing all subsequent shards with higher IDs, adding complexity to the process.
+
+In this NEP, we propose updating the ShardID semantics to allow for arbitrary identifiers. Although ShardIDs will remain integers, they will no longer be restricted to the `[0,n)` range, and they may appear in any order. The only requirement is that each ShardID must be unique. In practice, during resharding, the ID of a parent shard will be removed from the ShardLayout, and the new child shards will be assigned unique IDs - `max(shard_ids)+1` and `max(shard_ids)+2`.
```

Comment:
>that each ShardID must be unique

Unique in a given shard_layout or in the entire history of the blockchain? Just asking for myself

Reply:
Well in the past they were not unique so it's not going to be entire history. Going forward, yes, the new ones will be unique forever. (unless we run out of numbers and need to improvise ;) )